libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,203 @@
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <cmath>
+#include <algorithm>
+
+#include <catch2/catch.hpp>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+#include <ert/res_util/matrix_blas.hpp>
+
+#include <ert/analysis/update.hpp>
+#include <ert/analysis/std_enkf.hpp>
+
+namespace analysis {
+void run_analysis_update_without_rowscaling(analysis_module_type *module,
+                                            const bool_vector_type *ens_mask,
+                                            const meas_data_type *forecast,
+                                            obs_data_type *obs_data,
+                                            rng_type *shared_rng,
+                                            matrix_type *E, matrix_type *A);
+}
+
+const double a_true = 1.0;
+const double b_true = 5.0;
+
+struct model {
+    double a;
+    double b;
+
+    model(double a, double b) : a(a), b(b) {}
+
+    model(rng_type *rng) {
+        double a_std = 2.0;
+        double b_std = 2.0;
+        // Priors with bias
+        this->a = enkf_util_rand_normal(a_true + 0.5 * a_std, a_std, rng);
+        this->b = enkf_util_rand_normal(b_true - 0.5 * b_std, b_std, rng);
+    }
+
+    double eval(double x) const { return this->a * x + this->b; }
+};
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+
+    GIVEN("Fixed prior and measurements") {
+
+        analysis_module_type *enkf_module = analysis_module_alloc("STD_ENKF");
+        analysis_module_set_var(enkf_module, ENKF_TRUNCATION_KEY_, "1.0");
+        rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+
+        int ens_size = 100;
+
+        bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+        meas_data_type *meas_data = meas_data_alloc(ens_mask);
+        obs_data_type *obs_data = obs_data_alloc(1.0);
+
+        model true_model{a_true, b_true};
+
+        std::vector<model> ens;
+        for (int iens = 0; iens < ens_size; iens++)
+            ens.push_back(model(rng));
```

Comment:
Not that it matters - but here you can use `ens.emplace_back( rng ); ` to get in place construction.

Reply:
Nice, I needed to google this and came across [this nice explanation:](https://yasenh.github.io/post/cpp-diary-1-emplace_back/)

"...emplace_back directly takes constructor arguments for objects to be inserted. In other words, the emplacement function avoids constructing and destructing temporary objects. It will be much more efficient if developer try to insert large amount of objects or if each object is time consuming to create/destroy."

As you say, it does not matter in this case, but still nice to include it.

Thanks.